### PR TITLE
Add bias wave task to pysmurf-controller agent

### PR DESCRIPTION
**Pull request by bkelle**
adding in code for bias waves. Essentially copying the bias step code block and modifying for bias wave analysis.

## Description

adding in agent task for bias waves in similar form to bias step

## Motivation and Context

Needed to be able to run bias waves in nextline

### CODE REVIEW

File: socs/agents/pysmurf_controller/agent.py [hunk 1]
```diff
@@ -831,6 +831,112 @@ def publish_quantile_block(arr, name):
 
             return True, "Finished taking bias steps"
 
+    @ocs_agent.param('bgs', default=None)
+    @ocs_agent.param('kwargs', default=None)
```
Comment (jlashner) [inline socs/agents/pysmurf_controller/agent.py:835]: Needs param for the "tag" arg

File: socs/agents/pysmurf_controller/agent.py [hunk 2]
```diff
@@ -831,6 +831,112 @@ def publish_quantile_block(arr, name):
...
+    def take_bias_waves(self, session, params):
+        """take_bias_waves(kwargs=None, rfrac_range=(0.2, 0.9), tag=None)
+
+        **Task** - Takes bias_wave and saves the output filepath to the
+        session data object. See the `sodetlib bias wave docs page
+
+        NEED TO CREATE DOCS PAGE AND INSERT HERE
+
+        for more information on bias wave and what kwargs can be passed in.
```
Comment (jlashner) [inline socs/agents/pysmurf_controller/agent.py:844]: Lets leave this out until the docs page exists

**Review by jlashner — CHANGES_REQUESTED** [review]
Looks fine, a few inline comments. Also need to register the OCS operation at the bottom of the file.

I'd like this to be tested before merging, so we should work out when to do that.

File: socs/agents/pysmurf_controller/agent.py (hunk 1 above)
Comment (bkelle) [reply socs/agents/pysmurf_controller/agent.py:836]: fixed in latest commit

**Comment by bkelle** [comment]
> Looks good to me! a few inline comments. Also need to register the OCS operation at the bottom of the file.
> 
> I'd like this to be tested before merging, so we should work out when to do that.

I've made the requested changes, and registered the OCS operation at the end of the file! This should be good to test, I will interface with the site team to see when a good time for me to do that will be.

File: socs/agents/pysmurf_controller/agent.py [hunk 3]
```diff
@@ -833,6 +833,110 @@ def publish_quantile_block(arr, name):
...
+            # Resistance quantiles
+            publish_quantile_block(bwa.R0, 'Rtes')
+            publish_quantile_block(bwa.Si, 'resposnivity')
+            publish_quantile_block(bwa.Rfrac, 'Rfrac')
+
+            return True, "Finished taking bias waves"
+
+    @ocs_agent.param('bgs', default=None)
+    @ocs_agent.param('kwargs', default=None)
+    @ocs_agent.param('tag', default=None)
```
Comment (BrianJKoopman) [inline socs/agents/pysmurf_controller/agent.py:939]: Remove this extra 'tag' arg added to `overbias_tes()`.

**Review by BrianJKoopman — COMMENTED** [review]
Looks good to me, there was a stray decorator added to another task that should be removed (see below.)

Also, this bias wave code isn't merged upstream in sodetlib yet, so those tests are going to keep failing, and the build isn't going to work. That should be merged first.

File: socs/agents/pysmurf_controller/agent.py (hunk 3 above)
Comment (bkelle) [reply socs/agents/pysmurf_controller/agent.py:939]: fixed

**Comment by msilvafe** [comment]
> Looks good to me, there was a stray decorator added to another task that should be removed (see below.)
> 
> Also, this bias wave code isn't merged upstream in sodetlib yet, so those tests are going to keep failing, and the build isn't going to work. That should be merged first.

This was merged in sodetlib [here](https://github.com/simonsobs/sodetlib/pull/412) can we rerun ci tests somehow?

**Comment by BrianJKoopman** [comment]
Merging in the latest `main` would do it.

**Comment by msilvafe** [comment]
@BrianJKoopman this test failure doesn’t seem related to anything we added.

**Comment by BrianJKoopman** [comment]
> @BrianJKoopman this test failure doesn’t seem related to anything we added.

Yup, agreed. Not sure why this is cropping back up suddenly, but #643 should work around the issue here.

**Comment by BrianJKoopman** [comment]
Alright, #650 dropped the unrelated tests that are being flaky. Merge in the latest `main` when you're ready.

**Comment by BrianJKoopman** [comment]
Sorry, one more merge of the latest `main`. #652 drops the now failing tests.

**Review by msilvafe — APPROVED** [review]
Thanks, let's get this merged and running on the telescopes :)

**Review by jlashner — APPROVED** [review]

**Pull request merged**